tools/wake-unit/cli_options.cpp

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2023 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "../tools/wake/cli_options.h"
+
+#include <string.h>
+
+#include "unit.h"
+
+TEST(cli_options_basic) {
+  {
+    int argc = 2;
+    char *argv[] = {(char *)"wake", (char *)"--help", nullptr};
+
+    CommandLineOptions clo(argc, argv);
+
+    EXPECT_TRUE(clo.help);
+    EXPECT_FALSE(clo.init);
+    EXPECT_EQUAL(1, clo.argc);
+  }
+
+  {
+    int argc = 4;
+    char *argv[] = {(char *)"wake", (char *)"-v", (char *)"-x", (char *)"Unit", nullptr};
+
+    CommandLineOptions clo(argc, argv);
+
+    EXPECT_TRUE(clo.verbose);
+    EXPECT_EQUAL("Unit", clo.exec);
+    EXPECT_EQUAL(1, clo.argc);
+  }
+
+  {
+    int argc = 3;
+    char *argv[] = {(char *)"wake", (char *)"--failed", (char *)"--script", nullptr};
+
+    CommandLineOptions clo(argc, argv);
+
+    EXPECT_TRUE(clo.failed);
+    EXPECT_TRUE(clo.script);
+    EXPECT_EQUAL(1, clo.argc);
+  }
+}
+
+TEST(cli_options_target) {
+  int argc = 4;
+  char *argv[] = {(char *)"wake", (char *)"-v", (char *)"build", (char *)"default", nullptr};
+
+  CommandLineOptions clo(argc, argv);
+
+  EXPECT_TRUE(clo.verbose);
+  EXPECT_EQUAL(3, clo.argc);
+  EXPECT_EQUAL("build", clo.argv[1]);
+  EXPECT_EQUAL("default", clo.argv[2]);
+}
+
+// Timeline is done poorly and relies on positional args even though it only
+// has 3 possible values
+TEST(cli_options_timline) {
+  {
+    int argc = 2;
+    char *argv[] = {(char *)"wake", (char *)"--timeline", nullptr};
+
+    CommandLineOptions clo(argc, argv);
+
+    EXPECT_TRUE(clo.timeline);
+    EXPECT_EQUAL(1, clo.argc);
+  }
+
+  {
+    int argc = 3;
+    char *argv[] = {(char *)"wake", (char *)"--timeline", (char *)"file-accesses", nullptr};
+
+    CommandLineOptions clo(argc, argv);
+
+    EXPECT_TRUE(clo.timeline);
+    EXPECT_EQUAL(2, clo.argc);
+    EXPECT_EQUAL("file-accesses", clo.argv[1]);
+  }
+
+  {
+    int argc = 3;
+    char *argv[] = {(char *)"wake", (char *)"--timeline", (char *)"job-reflections", nullptr};
+
+    CommandLineOptions clo(argc, argv);
+
+    EXPECT_TRUE(clo.timeline);
+    EXPECT_EQUAL(2, clo.argc);
+    EXPECT_EQUAL("job-reflections", clo.argv[1]);
+  }
+
+  // This should be a cli error, but isn't
```

Comment:
There are only three valid "values" for `--timeline`
- nothing
- `job-reflections`
- `file-accesses`

Any other valid shouldn't be allowed, though that should probably be a `validation()` failure not a parse failure

Reply:
I'm deeply upset with optional fields. I just think they should just not exist. I don't think anyone is using that flag so if you wanted to fix that in a follow up to add an explicit option for "nothing" so that the flag wasn't optional I would be very happy.

Reply:
Yeah, I fully agree. It should go away but I want to figure out what "nothing" actually means and make sure that no one is relying on it. I think its indirectly depended on via the VSCode extension

Reply:
Even if its depended on I think we can replace it with an explicit option is what I was trying to say.

Reply:
right agreed, but what to call that option? and where to update with that new option?

Reply:
I see what you were saying now. Yeah I don't know where those things are needed but "default", "nil", "empty" or "nothing" would be good strawman options for now. We just have to go figure out where to insert that.